src/promptflow/promptflow/_sdk/_connection_provider/_local_connection_provider.py

```diff
@@ -13,4 +13,6 @@ def __init__(self):
         self._operations = ConnectionOperations()
 
     def get(self, name: str, **kwargs):
-        return self._operations.get(name, **kwargs)
+        # Connection provider here target for execution, so we always get with secrets.
+        with_secrets = kwargs.pop("with_secrets", True)
+        return self._operations.get(name, with_secrets=with_secrets, **kwargs)
```

Comment:
Do we have a test case to guard? To prevent from encountering the error again

Reply:
I believe live test can cover it.